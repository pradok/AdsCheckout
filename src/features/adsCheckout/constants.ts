export const actionTypes = {
  CHECKOUT_ADD_TO_CART: 'CHECKOUT/ADD_TO_CART',
};
