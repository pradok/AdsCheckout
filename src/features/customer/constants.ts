export const actionTypes = {
  CUSTOMER_DISCOUNTS_FETCH: 'CUSTOMER/DISCOUNTS_FETCH',
  CUSTOMER_DISCOUNTS_FETCH_SUCCESS: 'CUSTOMER/DISCOUNTS_FETCH_SUCCESS',
  CUSTOMER_DISCOUNTS_FETCH_ERROR: 'CUSTOMER/DISCOUNTS_FETCH_ERROR',
};
