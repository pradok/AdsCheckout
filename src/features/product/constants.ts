export const actionTypes = {
  PRODUCTS_COLLECTION_FETCH: 'PRODUCTS/COLLECTION_FETCH',
  PRODUCTS_COLLECTION_FETCH_SUCCESS: 'PRODUCTS/COLLECTION_FETCH_SUCCESS',
  PRODUCTS_COLLECTION_FETCH_ERROR: 'PRODUCTS/COLLECTION_FETCH_ERROR',
};
