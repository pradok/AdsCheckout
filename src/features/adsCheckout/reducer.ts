import {actionTypes as at} from './constants';
import {CartState, AddToCartAction} from './models';

export const initialState: CartState = {
  addedSKUs: [],
  quantityById: {
    classic: 0,
    standout: 0,
    premium: 0,
  },
};

export default (state = initialState, action: AddToCartAction) => {
  switch (action.type) {
    case at.CHECKOUT_ADD_TO_CART:
      const skus = {...state.quantityById};
      skus[action.payload] += 1;
      return {...state, quantityById: skus};
    default:
      return state;
  }
};
