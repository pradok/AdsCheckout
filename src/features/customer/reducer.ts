import {actionTypes as at} from './constants';
import {CustomerDiscountAction, CustomerDiscountsState} from './models';

export const initialState: CustomerDiscountsState = {
  discounts: false,
  isLoading: false,
  isFetched: false,
};

export default (state = initialState, action: CustomerDiscountAction) => {
  switch (action.type) {
    case at.CUSTOMER_DISCOUNTS_FETCH:
      return {...state, isLoading: true, isFetched: false, discounts: initialState.discounts};
    case at.CUSTOMER_DISCOUNTS_FETCH_SUCCESS:
      return {...state, isLoading: false, isFetched: true, discounts: action.payload};
    case at.CUSTOMER_DISCOUNTS_FETCH_ERROR:
      return {...state, ...initialState};
    default:
      return state;
  }
};
