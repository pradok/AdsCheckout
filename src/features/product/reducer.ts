import {actionTypes as at} from './constants';
import {ProductAction, ProductCollectionState} from './models';

export const initialState: ProductCollectionState = {
  collection: [],
  isLoading: false,
  isFetched: false,
};

export default (state = initialState, action: ProductAction) => {
  switch (action.type) {
    case at.PRODUCTS_COLLECTION_FETCH:
      return {...state, isLoading: true, isFetched: false, collection: initialState.collection};
    case at.PRODUCTS_COLLECTION_FETCH_SUCCESS:
      return {...state, isLoading: false, isFetched: true, collection: action.payload};
    case at.PRODUCTS_COLLECTION_FETCH_ERROR:
      return {...state, ...initialState};
    default:
      return state;
  }
};
