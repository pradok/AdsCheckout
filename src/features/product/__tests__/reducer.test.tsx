import reducer, {initialState} from '../reducer';
import {actionTypes as at} from '../constants';

describe('product reducers', () => {
  it(`Actioon ${at.PRODUCTS_COLLECTION_FETCH} should return state`, () => {
    expect(
      reducer(initialState,
        {
          type: at.PRODUCTS_COLLECTION_FETCH,
        }
      )
    ).toMatchSnapshot();
  });

  it(`Actioon ${at.PRODUCTS_COLLECTION_FETCH_SUCCESS} should return a state with list of ad products`, () => {
    expect(
      reducer(initialState,
        {
          type: at.PRODUCTS_COLLECTION_FETCH_SUCCESS,
          payload: [
            {
              id: 'classic',
              title: 'Classic Ad',
              price: 269.99,
            },
            {
              id: 'standout',
              title: 'Standout Ad',
              price: 322.99,
            },
          ],
        }
      )
    ).toMatchSnapshot();
  });

  it(`Actioon ${at.PRODUCTS_COLLECTION_FETCH_ERROR} should return state`, () => {
    expect(
      reducer(initialState,
        {
          type: at.PRODUCTS_COLLECTION_FETCH_ERROR,
        }
      )
    ).toMatchSnapshot();
  });

});
