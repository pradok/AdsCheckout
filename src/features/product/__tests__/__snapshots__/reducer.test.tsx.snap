// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product reducers Actioon PRODUCTS/COLLECTION_FETCH should return state 1`] = `
Object {
  "collection": Array [],
  "isFetched": false,
  "isLoading": true,
}
`;

exports[`product reducers Actioon PRODUCTS/COLLECTION_FETCH_ERROR should return state 1`] = `
Object {
  "collection": Array [],
  "isFetched": false,
  "isLoading": false,
}
`;

exports[`product reducers Actioon PRODUCTS/COLLECTION_FETCH_SUCCESS should return a state with list of ad products 1`] = `
Object {
  "collection": Array [
    Object {
      "id": "classic",
      "price": 269.99,
      "title": "Classic Ad",
    },
    Object {
      "id": "standout",
      "price": 322.99,
      "title": "Standout Ad",
    },
  ],
  "isFetched": true,
  "isLoading": false,
}
`;
